Adding SSVEP datasets

sylvchev opened the pull request:
Here is a draft to add more SSVEP datasets

sweetascherri reviewed (approved):
pipelines/CCA-SSVEP.yml

sylvchev commented:
We may need to update the minimal SSVEP example, to reflect that the default number of classes for `SSVEP` is now `None`

sylvchev reviewed (commented):
Remarks here will be corrected in the next commits.

Div12345 commented on moabb/datasets/ssvep_mamem1.py:
```diff
@@ -167,7 +170,7 @@ def _get_single_subject_data(self, subject):
             montage = make_standard_montage("GSN-HydroCel-256")
             raw.set_montage(montage)
-            sessions[session_name] = raw
+            sessions[session_name] = {'run_0': raw}
```
Why do you save it so?

sylvchev replied:
The EEG are stored in memory with MNE format, in a nested dict for each subject, with runs stored inside sessions elements. The rationale is that sessions are separate recordings that requires to setup the EEG cap and calibrating the system, while runs (or blocks) are recordings that are separated by breaks, subjects keeping their cap.

Div12345 commented:
@sylvchev If you can make a review of the code for the MAMEM3, this PR should be ready to merge

sylvchev commented:
I'm factorizing the three mamem codes and we are good !

sylvchev commented:
I corrected some problem for loading dataset 3. I combined all the code for MAMEM datasets in one file. The class `BaseMAMEM` contains all the code to load all the datasets, the specifics for MAMEM3 are included. I also added two examples, one for MAMEM1 and one for all the 3 MAMEM.
